extensions/amp-story/1.0/parallax-fx.js

```diff
@@ -0,0 +1,352 @@
+          const layerSpacing = Number(
+            page.getAttribute(LAYER_SPACING_ATTR) ||
+              this.storyLayerSpacing_ ||
+              DEFAULT_LAYER_SPACING
+          );
+          const farthestScale = Number(
+            page.getAttribute(FARTHEST_SCALE_ATTR) ||
+              this.storyFarthestScale_ ||
+              DEFAULT_FARTHEST_SCALE
+          );
+          const nearestScale = Number(
+            page.getAttribute(NEAREST_SCALE_ATTR) ||
+              this.storyNearestScale_ ||
+              DEFAULT_NEAREST_SCALE
+          );
+
+          this.initLayers_(
+            layers,
+            mode,
+            layerSpacing,
+            farthestScale,
+            nearestScale
+          );
+        });
+    });
+  }
+
+  /**
+   * Initializes the page's layers by giving them the appropriate translation and scaling
+   * @param {!Array<!Element>} layers
+   * @param {string} mode
+   * @param {number} layerSpacing
+   * @param {number} farthestScale
+   * @param {number} nearestScale
+   */
+  initLayers_(layers, mode, layerSpacing, farthestScale, nearestScale) {
+    // Loop through the layers in the page and assign a z-index following
+    // DOM order (manual override will be added in the future)
+    let order = 1;
+
+    const layerZIndexOffset =
+      mode == ParallaxModes.DEPTH
+        ? layers.length
+        : mode == ParallaxModes.CENTER
+        ? layers.length / 2
+        : 0;
+
+    layers.map(layer => {
+      this.vsync_.mutate(() => {
+        const translation = `translateZ(${(order - layerZIndexOffset) *
+          layerSpacing *
+          (PERSPECTIVE / 50)}px)`;
+        const scale = `scale(${mapRange(
+          order,
+          1,
+          layers.length,
+          farthestScale,
+          nearestScale
+        )})`;
+        setImportantStyles(layer, {
+          contain: 'none',
+          overflow: 'visible',
+          transform: `${translation} ${scale}`,
+        });
+        order++;
+      });
+    });
+  }
+
+  /**
+   * Discovers and returns all pages inside the story
+   * @return {!Array<!Element>}
+   */
+  getPages() {
+    return toArray(this.story_.querySelectorAll('amp-story-page')).map(page =>
+      dev().assertElement(page)
+    );
+  }
+
+  /**
+   * Discovers and returns all layers inside a page
+   * @param {!Element} page
+   * @return {!Array<!Element>}
+   */
+  getLayers(page) {
+    return toArray(page.querySelectorAll(`amp-story-grid-layer`)).map(layer =>
+      dev().assertElement(layer)
+    );
+  }
+
+  /**
+   * Calculate the smoothed change in position using the device orientation
+   * change event then update each of the layers will its calculated position.
+   * @param {Event} event
+   * @param {!Array<!ParallaxPage>} pages
+   * @private
+   */
+  parallaxOrientationMutate_(event, pages) {
+    const window = this.win_;
+    const {screen} = window;
+    let {gamma, beta} = event;
+
+    let angle;
```

Comment:
Better to use a const here rather than using a `let`, since that implies this could change later in the function.

```javascript
const angle = 'orientation' in screen ? screen.orientation.angle : window.orientation;
```